### ISSUE TITLE
Sometimes fixed elements have wrong offset

### ISSUE THREAD

**Issue by muxin** [ampproject/amphtml#5406]
Some position fixed elements could have wrong offset because fixed layer doesn't adjust the top correctly. It seems that the top should add `paddingTop`. This bug could happen in non-embedded mode and embedded mode, android and ios.

**Comment by jridgewell** [comment]
Are you talking about the layoutBox offsets?

**Comment by ericlindley-g** [comment]
I'd be curious to see an example of this to gauge severity

**Comment by muxin** [comment]
Not the layoutBox offset, it's the `element.style.top` value.
The bug happens in some extreme case when amp is embedded in viewer.
In https://github.com/ampproject/amphtml/blob/master/src/service/fixed-layer.js#L275, sometimes the `top` is explicitly set by the user but we thinks `isImplicitAuto == true`. Then we don't calculate the adjusted `top` for the fixed element.

**Comment by muxin** [comment]
We compares `offsetTop` values with and without `style.top = auto` to determine whether an element is `top: auto`. In some extreme case, the `offsetTop` is equal to imagined `auto-offset` but that element is actually not `top:auto`.
@dvoytenko First of all, I wonder why we didn't do `calc(paddingTop + X)` when `top != auto`?

**Comment by muxin** [comment]
For example, it is broken at https://yuxichen.herokuapp.com/examples/article-fixed-header.amp.max.html#paddingTop=54, but works fine at https://yuxichen.herokuapp.com/examples/article-fixed-header.amp.max.html#paddingTop=50

**Comment by dvoytenko** [comment]
An element with `top == auto` most likely has `bottom == X`. If we offset `top`, we'd essentially override `bottom`position. E.g. a footer element that was positioned flush with the bottom of the viewport will be instead positioned from the top. We have to calculate our own version of `top:auto` because there's no other way to test whether the user set an explicit `top` or not. This is because `.x {top: auto}` CSS in most of browsers will be resolved as `getComputedStyle(el).top == #px`.

The bottom line here is this: if the issue indeed is that the `auto` positioning coincided here with a declared non-auto position in CSS - this issue is ok to ignore.

**Comment by muxin** [comment]
OK, cool. This bug only happens in extreme cases. Closing the issue.
